src/annotator/highlighter.ts

```diff
@@ -306,6 +307,38 @@ export function removeHighlights(highlights: HighlightElement[]) {
   }
 }
 
+/**
+ * Focus or un-focus an individual SVG highlight element.
+ *
+ * When focusing an SVG highlight, make sure it is not obscured by other SVG
+ * highlight elements. As SVG highlights are siblings, this can be accomplished
+ * by putting the highlight at the end the set of highlights contained its
+ * parent. SVG highlight elements are cloned instead of moved so that their
+ * original stacking (nesting) order is not lost when later unfocused. A data
+ * attribute is added to associate the original SVG highlight element with its
+ * clone.
+ */
+function setSVGHighlightFocused(svgEl: SVGElement, focused: boolean) {
+  const parent = svgEl.parentNode as SVGElement;
+  const focusedId = svgEl.getAttribute('data-focused-id');
+  if (focused) {
+    if (!focusedId) {
+      svgEl.setAttribute('data-focused-id', generateHexString(8));
+    }
+    const focusedHighlight = svgEl.cloneNode() as SVGRectElement;
+    focusedHighlight.classList.add('is-focused');
+    parent.append(focusedHighlight);
```

Comment:
If a cluster ordering update is pending at the point when the focused highlight is created, the ordering update which fires later may put the highlight somewhere below un-focused highlights in the stacking order. We need to ensure that focused highlights always end up on top.

I was able to reproduce an issue where nested highlights were not always rendered correctly when focused as follows:

1. Create some overlapping highlights in a long document (I use http://localhost:3000/pdf/gatsby for most testing)
2. Scroll a long distance away so that the pages are un-rendered
3. Click on annotation card for one of the annotations

<img width="962" alt="Wrong focus highlight" src="https://user-images.githubusercontent.com/2458/203512236-a5b25346-c662-4a8e-892d-35de405c307a.png">